multiclient/client.go

```diff
@@ -657,46 +657,69 @@ func (mc *Client) SubscribeNewHead(ctx context.Context, ch chan<- *Header) (ethe
 	}), nil
 }
 
-func (mc *Client) subscribeNewHead(ctx context.Context, wg *sync.WaitGroup, url string, ch chan<- *Header) error {
+func (mc *Client) subscribeNewHead(ctx context.Context, wg *sync.WaitGroup, id uint64, ch chan<- *Header) error {
+	logger := log.New("id", id)
 	defer wg.Done()
 
 	for {
-		rc := mc.rpcClientMap.Get(url)
-		if rc == nil {
-			log.Trace("EthClient has been removed", "url", url)
+		rc, existed := mc.rpcClientMap.GetById(id)
```

Comment:
How about return url?
`url` is more useful than `id` to logging.

Reply:
OK